src/components/forms/live-input-hoc.jsx

```diff
@@ -0,0 +1,60 @@
+import bindAll from 'lodash.bindall';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+/**
+ * Higher Order Component to manage inputs that submit on change and <enter>
+ * @param {React.Component} Input text input that consumes onChange, onBlur, onKeyPress
+ * @returns {React.Component} Live input that calls onSubmit on change and <enter>
+ */
+export default function (Input) {
+    class LiveInput extends React.Component {
+        constructor (props) {
+            super(props);
+            bindAll(this, [
+                'handleChange',
+                'handleKeyPress',
+                'handleFlush'
+            ]);
+            this.state = {
+                value: null
+            };
+        }
+        handleKeyPress (e) {
+            if (e.key === 'Enter') {
+                this.handleChange(e);
+                e.target.blur();
+            }
+        }
+        handleFlush () {
+            this.setState({value: null});
+        }
+        handleChange (e) {
+            const isNumeric = typeof this.props.value === 'number';
+            const validatesNumeric = isNumeric ? !isNaN(e.target.value) : true;
```

Comment:
I think you need max value checking here. When you type in a huge number and press enter, the value in the field snaps back down to 400 but the image's stroke width doesn't. It looks like for huge values, the canvas just gives up.

Reply:
@fsih great point – should we add this for the buffered input too, while I'm at it? /cc @paulkaplan

Reply:
It looks like the buffered inputs in the sprite info panel are handling max values OK already?

Reply:
@fsih ah ok – odd, since it's the same functionality just in a different method. I'll take a look...

Reply:
They're probably handling it outside the component, and it's easier since it doesn't have a "live" factor. But yeah @paulkaplan would know better